plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -55,11 +61,17 @@ export class TeamManager {
 
         if (!this.eventNamesCache.get(team.id)?.has(event)) {
             await this.db.postgresQuery(
-                `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id) VALUES ($1, $2, NULL, NULL, $3) ON CONFLICT DO NOTHING`,
-                [new UUIDT().toString(), event, team.id],
+                `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id, last_seen_at, created_at) VALUES ($1, $2, NULL, NULL, $3, $4, NOW()) ON CONFLICT DO NOTHING`,
+                [new UUIDT().toString(), event, team.id, eventTimestamp],
                 'insertEventDefinition'
             )
             this.eventNamesCache.get(team.id)?.add(event)
+        } else {
```

Comment:
o_o this is likely not a good idea - great way to create even more load for the postgres database and cause us to fall over.

Note that this is done every single event ingestion - it really doesn't need to be that frequent.

Reply:
Thanks for the early feedback, don't want to spend any more time going down the wrong direction. That was my first thought but I didn't want to add another background job that requires going through the entire list of events as we've had problems with that in the past (though it could be greatly optimised). 

How about relying on the caching mechanism we have to see if the event definition has been created and only update when 30min-1hr has passed?

Reply:
I guess the true solution is replacing `ASYNC_EVENT_PROPERTY_USAGE` with a plugin-server-based counter that updates `volume_30_day` on the fly, syncing data to postgres every 10-60 sec or so.

Reply:
I'll keep the `last_seen_at` as based on @clarkus's feedback, it'll be helpful to tell this information to the user (which `volume_30_day` wouldn't provide. I have an idea on how to improve the `ASYNC_EVENT_PROPERTY_USAGE` method, but for a separate PR.

Reply:
Syill think this is a bad idea - pretty sure this will blow up ingestion one way or another sooner or later - either queries or memory wise. 

Gonna cc @yakkomajuri here.

Reply:
Thanks for the pushback here, will wait for @yakkomajuri's input before moving forward.

Reply:
An alternative to prevent blowing up the DB is extending the threshold, maybe just update once a day instead of every 15 minutes, but unsure about memory. I'm worried that doing it via an async job will be worse, we'd have to go almost through all possible event names.

Reply:
If you `SELECT DISTINCT (team_id, event) FROM events` once per day in a cronjob + insert the same timestamp for all rows the query wouldn't be too bad.

Reply:
Some thoughts (trying to limit the amount of time I brainstorm on this):

1. Agree with @macobo, let's be careful with ingestion-triggered Postgres queries, particularly those to supplement functionality that isn't core to ingestion (as opposed to creating persons and distinct IDs for example).
2. A cronjob sounds like a it could be a good approach to me. Also, as you're planning to backfill, you'd only need to look through the events seen on that given day (use `_timestamp` for that), so might not be that bad. 
3. There could potentially be another solution here if you're feeling adventurous. You could leverage CH materialized views to do this. Although in that case you couldn't pull all the data you need from the event definition Postgres row and would have to hit CH.

I'd say probably test out what a daily cronjob would look like and see how it might perform. It could also help with performance to remove granularity here and store `last_seen_at` to the "nearest day", so you don't need to pull timestamps and instead just do something like:

```
SELECT DISTINCT (team_id, event) FROM events WHERE _timestamp > today()
```

Final point is that it could be worth starting this out as an enterprise feature to test it out.